sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj

```diff
@@ -12,12 +12,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <!-- Remove the project reference and restore the package reference when the next version of the Event Hubs core is published. -->
-    <ProjectReference Include="$(MSBuildThisFileDirectory)..\..\Azure.Messaging.EventHubs\src\Azure.Messaging.EventHubs.csproj" />
-    <!--
-    <PackageReference Include="Azure.Messaging.EventHubs" />
-    -->
-
+    <PackageReference Include="Azure.Messaging.EventHubs" VersionOverride="5.1.0-preview.1" />
```

Comment:
Is there a reason to pin this version to a preview? Does this make it better then doing the ProjectReference? It would seem that you will still likely need to ship both of these together.

Reply:
Yes; this version has a dependency on the preview and is not compatible with the GA package.  These packages are expected to evolve independently and we do not want to ship a project reference.  Once the preview features GA, the version override will no longer be needed and will be removed.

Reply:
I highly recommend that we continue to ship these together until we get a GA version we can depend on. They are already coupled together such that Processor cannot ship a GA until the base package it depends on also ships a GA. 

For reference this issue was found by helping @danieljurek investigate an issue with the smoke tests https://dev.azure.com/azure-sdk/internal/_build/results?buildId=349982&view=logs&j=49400574-4c50-5d31-7174-432d6b5afe93&t=45f8ce64-9106-5354-2ac2-2d79cb0e8b7b&l=129. While the error can probably be ignored we are trying to figure out if using the VersionOverride is the best strategy for this case.

Reply:
I've no issue using a project reference in the repository for development activities and nightly runs.